react/features/toolbox/components/web/OverflowMenuLiveStreamingItem.js

```diff
@@ -1,66 +1,110 @@
  */
-type Props = {
-
+class OverflowMenuLiveStreamingItem extends Component {
     /**
-     * The callback to invoke when {@code OverflowMenuLiveStreamingItem} is
-     * clicked.
+     * Default values for {@code OverflowMenuLiveStreamingItem}
+     * component's properties.
+     *
+     * @static
      */
-    onClick: Function,
+    static defaultProps = {
+        tooltipPosition: 'left',
+        disabled: false
+    };
 
     /**
-     * The current live streaming session, if any.
+     * The type of the React {@code Component} props of
+     * {@link OverflowMenuLiveStreamingItem}.
      */
-    session: ?Object,
+    static propTypes = {
 
-    /**
-     * Invoked to obtain translated strings.
-     */
-    t: Function
-};
+        /**
+         * Whether menu item is disabled or not.
+         */
+        disabled: PropTypes.bool,
+
+        /**
+         * The callback to invoke when {@code OverflowMenuLiveStreamingItem} is
+         * clicked.
+         */
+        onClick: Function,
+
+        /**
+         * The current live streaming session, if any.
+         */
+        session: PropTypes.object,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: Function,
+
+        /**
+         * The text to display in the tooltip.
+         */
+        tooltip: PropTypes.string,
+
+        /**
+         * From which direction the tooltip should appear, relative to the
+         * button.
+         */
+        tooltipPosition: PropTypes.string
+    };
 
-/**
- * React {@code Component} for starting or stopping a live streaming of the
- * current conference and displaying the current state of live streaming.
- *
- * @extends Component
- */
-class OverflowMenuLiveStreamingItem extends Component<Props> {
     /**
      * Implements React's {@link Component#render()}.
      *
      * @inheritdoc
      * @returns {ReactElement}
      */
     render() {
-        const { onClick, session, t } = this.props;
+        const { disabled, onClick, session, t, tooltip, tooltipPosition }
+            = this.props;
 
         const translationKey = session
             ? 'dialog.stopLiveStreaming'
             : 'dialog.startLiveStreaming';
 
-        return (
-            <li
-                aria-label = { t('dialog.accessibilityLabel.liveStreaming') }
-                className = 'overflow-menu-item'
-                onClick = { onClick }>
-                <span className = 'overflow-menu-item-icon'>
-                    <i className = 'icon-public' />
-                </span>
+        let className = 'overflow-menu-item';
+
+        className += disabled ? ' disabled' : '';
+
+        const button = (// eslint-disable-line no-extra-parens
+            <span>
                 <span className = 'profile-text'>
                     { t(translationKey) }
                 </span>
                 <span className = 'beta-tag'>
                     { t('recording.beta') }
                 </span>
+            </span>
```

Comment:
I'm trying to get confirmation from Emil now on how long this beta tag will last. If it's not too much longer I won't do anything but if it's gonna be around for like a year then I'll refactor OverflowMenuItem to take in the beta tag.

Reply:
Emil said it'll be around for a while so I'll refactor it in another commit.